packages/cw-orch-core/src/environment/queriers/wasm.rs

```diff
@@ -48,3 +48,11 @@ pub trait WasmQuerier: Querier {
         salt: cosmwasm_std::Binary,
     ) -> Result<String, Self::Error>;
 }
+
+pub trait AsyncWasmQuerier: Querier {
+    async fn smart_query<Q: Serialize, T: DeserializeOwned>(
```

Comment:
What about those clippy warnings ? Can we use async traits fn in traits now ? Nice

Reply:
https://users.rust-lang.org/t/async-in-public-trait/108400

Reply:
Yeah so we should add send bound for sure (as that enables multi-threaded application) 

Also we can use this crate mentioned in that post to create a variant with / without the Send constraint: https://crates.io/crates/trait-variant

We might need this for the async -> sync wrapping that we do (i.e. use the non-send trait in that context). But the compiler will help us figure that out.